syntax = "proto3";

message LoginRequest{
  string password = 1;
  string login = 2;
}

message LoginResponse{
  string role_id = 1;
  string first_name = 2;
}

message HeaderRequest{
  string login = 1;
}

message HeaderResponse{
  string user_name = 1;
  string role_id = 2;
}

message RegisterUserRequest{
  string current_user = 1;
  string first_name = 2;
  string last_name = 3;
  string email = 4;
  string phone_number = 5;
  string role = 6;
}

message RegisterUserResponse{
  bool user_created = 1;
  bool email_send_to_user = 2;
}

message RegisterSchoolRequest{
  string current_user = 1;
  string name = 2;
  string email = 3;
  string phone_number = 4;
  string address = 5;
}

message RegisterSchoolResponse{
  bool school_created = 1;
  bool email_send_to_school = 2;
}

service AuthService{
  rpc Login (LoginRequest) returns (LoginResponse);
  rpc Header (HeaderRequest) returns (HeaderResponse);
  rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse);
  rpc RegisterSchool (RegisterSchoolRequest) returns (RegisterSchoolResponse);
}